requests_cache/core.py

```diff
@@ -95,50 +136,131 @@ def send(self, request, **kwargs):
         if response is None:
             return self.send_request_and_cache_response(request, cache_key, **kwargs)
 
-        if self._cache_expire_after is not None:
-            is_expired = datetime.utcnow() - timestamp > self._cache_expire_after
-            if is_expired:
-                if not self._return_old_data_on_error:
-                    self.cache.delete(cache_key)
-                    return self.send_request_and_cache_response(request, cache_key, **kwargs)
-                try:
-                    new_response = self.send_request_and_cache_response(request, cache_key, **kwargs)
-                except Exception:
+        if getattr(response, 'expiration_date', None) is not None:
+            now = datetime.now(timezone.utc)
+            is_expired = now > response.expiration_date
+        else:
+            is_expired = False
+
+        cache_invalid = response.expire_after != self._request_expire_after and self._request_expire_after != 'default'
+        if cache_invalid or is_expired:
+            if not self._return_old_data_on_error:
+                self.cache.delete(cache_key)
+                return self.send_request_and_cache_response(request, cache_key, **kwargs)
+            try:
+                new_response = self.send_request_and_cache_response(request, cache_key, **kwargs)
+            except Exception:
+                return response
+            else:
+                if new_response.status_code not in self._cache_allowable_codes:
                     return response
-                else:
-                    if new_response.status_code not in self._cache_allowable_codes:
-                        return response
-                    return new_response
+                return new_response
 
         # dispatch hook here, because we've removed it before pickling
         response.from_cache = True
         response.cache_date = timestamp
         response = dispatch_hook('response', request.hooks, response, **kwargs)
         return response
 
-    def send_request_and_cache_response(self, request, cache_key, **kwargs):
-        response = super().send(request, **kwargs)
-        if response.status_code in self._cache_allowable_codes:
-            self.cache.save_response(cache_key, response)
-        response.from_cache = False
-        response.cache_date = None
-        return response
-
-    def request(self, method, url, params=None, data=None, **kwargs):
+    def request(self, method, url, params=None, data=None, expire_after='default', **kwargs):
```

Comment:
The option of setting `expires_after='cached'` may not be necessary. I can think of two different ways a user would want to handle expiration:
1. Set the absolute expiration time when a response is cached, and remove the cached response after that time.
2. Override that expiration time later on; for example, "I want to remove anything in the cache older than 2 days, regardless of its original expiration time."

Both cases can be accomplished without adding and checking for a value of `'cached'`. I would prefer to leave that out for the sake of reducing code complexity, unless you or someone has a specific use case for it. Let me know if I'm missing something there.

The value `'default'` may also be a bit confusing to some users, since in most caching systems (like a browser cache), the "default" behavior is to set the expiration time when the cache item is created, not when it's accessed. I think it's fine to keep that as it is i your PR for now, but I may want to think about that a bit more before the next release.

Reply:
`'default'` should set the expiration time when it is created, or am I missing something here?

I agree that removing 'cached' reduces the code complexity a lot and I only implemented it at the end. WIll definitely be removed.

Reply:
I removed the 'cached' keyword, but kept the term 'default' for now. If I am not mistaken, the expiration date is set when the item is saved:
https://github.com/reclosedev/requests-cache/pull/177/files#diff-ae7b3e168473eb551467ffabd378f23ea2a09a81c36281b8cf8ea394448bb5feR245

Reply:
I see, I misunderstood what 'default' did. Looks good.